scripts/liquidation_bot/liquidateUnderwaterBorrowers.ts

```diff
@@ -19,49 +20,45 @@ export interface Asset {
   scale: bigint;
 }
 
-const daiPool = {
-  tokenAddress: '0x6B175474E89094C44Da98b954EedeAC495271d0F',
-  poolFee: 100
-};
-
-async function sendFlashbotsPrivateTransaction(
-  txn: PopulatedTransaction,
-  flashbotsProvider: FlashbotsBundleProvider
-) {
-  const privateTx = {
-    transaction: txn,
-    signer: flashbotsProvider.getSigner(),
-  };
-  await flashbotsProvider.sendPrivateTransaction(privateTx);
-}
-
-async function sendTxn(
-  txn: PopulatedTransaction,
-  signerWithFlashbots: SignerWithFlashbots
-) {
-  if (signerWithFlashbots.flashbotsProvider) {
-    googleCloudLog(LogSeverity.INFO, 'Sending a private txn');
-    await sendFlashbotsPrivateTransaction(txn, signerWithFlashbots.flashbotsProvider);
-  } else {
-    googleCloudLog(LogSeverity.INFO, 'Sending a public txn');
-    await signerWithFlashbots.signer.sendTransaction(txn);
+const flashLoanPools = {
+  'mainnet': {
+    // DAI pool
+    tokenAddress: '0x6B175474E89094C44Da98b954EedeAC495271d0F',
+    poolFee: 100
+  },
+  'goerli': {
+    // WETH pool
+    tokenAddress: '0xB4FBF271143F4FBf7B91A5ded31805e42b2208d6',
+    poolFee: 3000
   }
-}
+};
 
 async function attemptLiquidation(
   liquidator: Liquidator,
   targetAddresses: string[],
-  signerWithFlashbots: SignerWithFlashbots
+  signerWithFlashbots: SignerWithFlashbots,
+  network: string
 ) {
   try {
     googleCloudLog(LogSeverity.INFO, `Attempting to liquidate ${targetAddresses} via ${liquidator.address}`);
-    const txn = await liquidator.populateTransaction.initFlash({
+    const flashLoanPool = flashLoanPools[network];
+    const calldata = {
       accounts: targetAddresses,
-      pairToken: daiPool.tokenAddress,
-      poolFee: daiPool.poolFee
+      pairToken: flashLoanPool.tokenAddress,
+      poolFee: flashLoanPool.poolFee
+    };
+    // XXX set appropriate gas price...currently we are overestimating slightly to be safe
+    // XXX also factor in gas price to profitability
+    const txn = await liquidator.populateTransaction.initFlash(calldata, {
+      gasLimit: Math.ceil(1.1 * (await liquidator.estimateGas.initFlash(calldata)).toNumber()),
+      gasPrice: Math.ceil(1.1 * (await hre.ethers.provider.getGasPrice()).toNumber()),
     });
-    await sendTxn(txn, signerWithFlashbots);
-    googleCloudLog(LogSeverity.INFO, `Successfully liquidated ${targetAddresses} via ${liquidator.address}`);
+    const success = await sendTxn(txn, signerWithFlashbots);
+    if (success) {
+      googleCloudLog(LogSeverity.INFO, `Successfully liquidated ${targetAddresses} via ${liquidator.address}`);
+    } else {
+      googleCloudLog(LogSeverity.ALERT, `Failed to liquidate ${targetAddresses} via ${liquidator.address}`);
```

Comment:
is it possible to add a stack trace or error message for a failed transaction?

Reply:
maybe it's already being logged on ln 44 of `sendTransaction`?

Reply:
Flashbots won't include transactions that revert, so `googleCloudLog(LogSeverity.INFO, 'Block passed without inclusion')` would get logged. Transactions that revert deterministically won't even make it to that point because gas estimation fails when constructing the txn, so you just end up with the alert log in the catch block below.